PVSSpringBoot/src/main/java/com/example/PVSSpringBoot/services/PetManagementService.java

```diff
@@ -0,0 +1,73 @@
+package com.example.PVSSpringBoot.services;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.example.PVSSpringBoot.Entities.Pet;
+import com.example.PVSSpringBoot.repositories.PetRepository;
+
+import java.sql.Date;
+import java.time.LocalDate;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class PetManagementService {
+    @Autowired
+    PetRepository repo;
+
+    //add a pet
+    public boolean addPet(Pet pet){
+        if(pet == null) return false;
+        if(repo.existsInDB(pet.getName(),pet.getType(),pet.getBreed())) return false;
```

Comment:
when a new pet is added the user does not assign an id to it but rather asssigned in DB using a sequence so the query should be done by other parameters (name,type,breed) - a tuple of attributes that is hard to be replicated unless it was the same animal "fairly unique"-
could add date of birth to that tuple and it would be totally unique